python/lsst/ts/mtrotator/rotator_csc.py

```diff
@@ -519,13 +548,134 @@ async def do_trackStart(self, data: salobj.BaseMsgType) -> None:
+        """Lock the rotator motion.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+            Data of the SAL message.
+
+        Raises
+        ------
+        `salobj.ExpectedError`
+            When the motion is failed to stop.
+        """
+
+        if self._is_motion_locked:
+            return
+
+        await self.cmd_lockMotion.ack_in_progress(data, timeout=self.COMMAND_TIMEOUT)
+
+        await self._pub_event_motion_lock_state(
+            1, data.private_identity
+        )  # Use MotionLockState.LOCKING
+
+        # Workaround the mypy check
+        assert self.client is not None
+
+        # Stop the motion when it is not stationary
+        if self.is_controller_enabled() and (
+            self.client.telemetry.enabled_substate != EnabledSubstate.STATIONARY
+        ):
+            await self.run_command(
+                code=enums.CommandCode.SET_ENABLED_SUBSTATE,
+                param1=enums.SetEnabledSubstateParam.STOP,
+            )
+
+            max_attempt = 10
+            attempt = 0
+            while (
+                self.client.telemetry.enabled_substate != EnabledSubstate.STATIONARY
+            ) and (attempt < max_attempt):
+                await asyncio.sleep(1.0)
+                attempt += 1
+
+            if attempt >= max_attempt:
+                raise salobj.ExpectedError(f"Cannot stop the motion with {attempt=}")
+
+        # Standby the Copley drives
+        if self.is_controller_enabled():
+            await self.standby_controller()
+
+        self._is_motion_locked = True
+
+        await self._pub_event_motion_lock_state(
+            3, data.private_identity
+        )  # Use MotionLockState.LOCKED
+
+    async def _pub_event_motion_lock_state(
+        self, lock_state: int, identity: str
+    ) -> None:
+        """Publish the event of motion lock state.
+
+        Parameters
+        ----------
+        lock_state : `int`
+            Lock state.
+        identity : `str`
+            Identity of the entity that locked/unlocked.
+        """
+
+        # TODO: Remove this method at the ts_xml v23.0.0. It is not supported
+        # at the current v22.1.0.
+        if hasattr(self, "evt_motionLockState"):
+            await self.evt_motionLockState.set_write(
+                lockState=lock_state, identity=identity
+            )
+
+    def is_controller_enabled(self) -> bool:
+        """Controller is enabled or not.
+
+        Returns
+        -------
+        `bool`
+            True if the controller is enabled. False otherwise.
+        """
+
+        # Workaround the mypy check
+        assert self.client is not None
+
+        return self.client.telemetry.state == ControllerState.ENABLED
+
+    async def do_unlockMotion(self, data: salobj.BaseMsgType) -> None:
+        """Unlock the rotator motion.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+            Data of the SAL message.
+        """
+
```

Comment:
I think this command also needs an `self.assert_enabled()` here no?

Reply:
Added in the commit: [b4e39d1](https://github.com/lsst-ts/ts_mtrotator/pull/97/commits/b4e39d11dc643201ed9fc28fd9b89b0c45c65069). Thanks!